include/drivers/spi.h

```diff
@@ -142,6 +142,61 @@ struct spi_cs_control {
 	gpio_dt_flags_t		gpio_dt_flags;
 };
 
+/**
+ * @brief Initialize and get a pointer to a @p spi_cs_control from a
+ *        devicetree node identifier
+ *
+ * This helper is useful for initializing a device on a SPI bus. It
+ * initializes a struct spi_cs_control and returns a pointer to it.
+ * Here, @p node_id is a node identifier for a SPI device, not a SPI
+ * controller.
+ *
+ * Example devicetree fragment:
+ *
+ *         spi@... {
+ *                 cs-gpios = <&gpio0 1 GPIO_ACTIVE_LOW>;
+ *                 spidev: spi-device@0 { ... };
+ *         };
+ *
+ * Example usage:
+ *
+ *         SPI_CS_CONTROL_PTR_DT(DT_NODELABEL(spidev), 2)
+ *
+ *         // Returns a pointer to a spi_cs_control with:
+ *         //        gpio_dev = struct device* for 'gpio0'
+ *         //        gpio_pin = 1,
+ *         //        delay = 2,
+ *         //        gpio_dt_flags = GPIO_ACTIVE_LOW,
+ *
+ * @param node_id Devicetree node identifier for a device on a SPI bus
+ * @param delay_ The @p delay field to set in the @p spi_cs_control
+ * @return a pointer to the @p spi_cs_control structure
+ */
+#define SPI_CS_CONTROL_PTR_DT(node_id, delay_)				\
+	(&(struct spi_cs_control) {						\
```

Comment:
Note that this construct (taking the address of a temporary) is illegal in C++ when the value is passed as a function argument.  That shouldn't prevent it being used here, though, since we're not supposed to write drivers in C++.

It could be worth a comment someplace, though, because "not supposed to" doesn't equal "people don't want to".

FYI: [The C++ workaround](https://github.com/zephyrproject-rtos/zephyr/pull/24829#issuecomment-622977341) is:

    auto ctrl = *SPI_CS_CONTROL_PTR_DT(node_id, delay);
    rc = spi_function(..., &ctrl, ...);

Reply:
> FYI: [The C++ workaround](https://github.com/zephyrproject-rtos/zephyr/pull/24829#issuecomment-622977341) is:

I've added a comment with this recommendation. That must have been a fun excursion to figure out.

Reply:
Did not work for me with `static constexpr auto ctrl = *SPI_CS_CONTROL_PTR_DT(node_id, delay);`